x/superfluid/keeper/grpc_query.go

```diff
@@ -392,6 +392,41 @@ func (q Querier) EstimateSuperfluidDelegatedAmountByValidatorDenom(goCtx context
 	}, nil
 }
 
+func (q Querier) TotalDelegationByValidatorForAsset(goCtx context.Context, req *types.QueryTotalDelegationByValidatorForAssetRequest) (*types.QueryTotalDelegationByValidatorForAssetResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	var delegationsByValidator = []*types.Delegations{}
+	var intermediaryAccount types.SuperfluidIntermediaryAccount
+
+	intermediaryAccounts := q.Keeper.GetAllIntermediaryAccounts(ctx)
+	for _, intermediaryAccount = range intermediaryAccounts {
+		if intermediaryAccount.Denom != req.Denom {
+			continue
+		}
+		valAddr, err := sdk.ValAddressFromBech32(intermediaryAccount.ValAddr)
+		if err != nil {
+			return nil, err
+		}
+
+		val, found := q.Keeper.sk.GetValidator(ctx, valAddr)
+		if !found {
+			return nil, stakingtypes.ErrNoValidatorFound
+		}
+		equivalentAmountOSMO := q.Keeper.GetSuperfluidOSMOTokens(ctx, req.Denom, val.Tokens)
+		result := &types.Delegations{
+			ValAddr:        valAddr.String(),
+			AmountSfsd:     &val.Tokens,
+			OsmoEquivalent: &equivalentAmountOSMO,
+		}
```

Comment:
hrmm this seems wrong to me. Returning val.Tokens would return all the tokens on the validator, not the amount on the intermediary account. Instead I think we should be doing
```
delegation, found := suite.App.StakingKeeper.GetDelegation(suite.Ctx, expAcc.GetAccAddress(), valAddr)
```

and then be using delegation.shares to track how much is sfsd

Reply:
@mattverse I do not really understand why do we need to get amount of an intermediary account if we want to get assets of each validator for a specific denomination? I would be very thankful for further clarification, please

Reply:
other comments were addressed

Reply:
Awesome! Good question, here we want to return the amount superfluid staked, not the amount total staked on the validator right? Say for example valA has total of 100uosmo staked, 50uosmo staked natively, 50uosmo staked via superfluid staking . `val.Tokens` would be returning 100uosmo, whilst using getDelegation would be returning 50uosmo, only the amount that has been staked

Reply:
oh, that was an awesome example! Thank you! I will try to figure out how to fix that!

Reply:
@mattverse this comment was also addressed